bridges/EuronewsBridge.php

```diff
@@ -0,0 +1,209 @@
+<?php
+class EuronewsBridge extends BridgeAbstract
+{
+	const MAINTAINER = 'sqrtminusone';
+	const NAME = 'Euronews Bridge';
+	const URI = 'https://www.euronews.com/';
+	const CACHE_TIMEOUT = 600; // 10 minutes
+	const DESCRIPTION = 'Return articles from the "Just In" feed of Euronews.';
+
+	const PARAMETERS = array(
+		'' => array(
+			'lang' => array(
+				'name' => 'Language',
+				'type' => 'list',
+				'defaultValue' => 'euronews.com',
+				'values' => array(
+					'English' => 'euronews.com',
+					'French' => 'fr.euronews.com',
+					'German' => 'de.euronews.com',
+					'Italian' => 'it.euronews.com',
+					'Spanish' => 'es.euronews.com',
+					'Portuguese' => 'pt.euronews.com',
+					'Russian' => 'ru.euronews.com',
+					'Turkish' => 'tr.euronews.com',
+					'Greek' => 'gr.euronews.com',
+					'Hungarian' => 'hu.euronews.com',
+					'Persian' => 'per.euronews.com',
+					'Arabic' => 'arabic.euronews.com',
+					/* These versions don't have timeline.json */
+					// 'Albanian' => 'euronews.al',
+					// 'Romanian' => 'euronews.ro',
+					// 'Georigian' => 'euronewsgeorgia.com',
+					// 'Bulgarian' => 'euronewsbulgaria.com'
+					// 'Serbian' => 'euronews.rs'
+				)
+			),
+			'limit' => array(
+				'name' => 'Limit of items per feed',
+				'required' => true,
+				'type' => 'number',
+				'defaultValue' => 10,
+				'title' => 'Maximum number of returned feed items. Maximum 50, default 10'
+			),
+		)
+	);
+
+	public function collectData()
+	{
+		$limit = $this->getInput('limit');
+		$root_url = 'https://' . $this->getInput('lang');
+		$url = $root_url . '/api/timeline.json?limit=' . $limit;
+		$json = getContents($url);
+		$data = json_decode($json, true);
+
+		foreach ($data as $datum) {
+			$datum_uri = $root_url . $datum['fullUrl'];
+			$url_datum = $this->getContent($datum_uri);
+			$categories = array();
+			if (array_key_exists('program', $datum)) {
+				if (array_key_exists('title', $datum['program'])) {
+					$categories[] = $datum['program']['title'];
+				}
+			}
+			if (array_key_exists('themes', $datum)) {
+				foreach ($datum['themes'] as $theme) {
+					$categories[] = $theme['title'];
+				}
+			}
+			$item = array(
+				'uri' => $datum_uri,
+				'title' => $datum['title'],
+				'uid' => strval($datum['id']),
+				'timestamp' => $datum['publishedAt'],
+				'content' => $url_datum['content'],
+				'author' => $url_datum['author'],
+				'enclosures' => $url_datum['enclosures'],
+				'categories' => array_unique($categories)
+			);
+			$this->items[] = $item;
+		}
+	}
+
+	private function getContent($url)
+	{
+		try {
+			$html = getSimpleHTMLDOMCached($url);
+		} catch (Exception $e) {
+			// Every once in a while it fails with too many redirects
```

Comment:
Did you try setting `CURLOPT_MAXREDIRS` in the `opts` argument of `getSimpleHTMLDOMCached`? See documentation [here](https://github.com/RSS-Bridge/rss-bridge/blob/master/lib/contents.php#L270=).

Reply:
If I understand correctly, that won't fix the error. The Euronews server has a bug - sometimes it produces infinite redirection loops.

Setting `CURLOPT_MAXREDIRS` to `0` is not an option because redirects are necessary, `-1` can cause infinite loop, and the default value is `10`.

Also, this seems to be a pretty rare case. I've only caught it a few days into using this bridge.

Reply:
Oh that makes sense. Thanks for looking into it!